### DISCUSSION TITLE
I got blank object when I add an image to rectangle?

### DISCUSSION THREAD

**Discussion by ekaaditiac** [fabricjs/fabric.js#8815] [Q&A] [answered]
Thank you for the awesome fabricjs canvas, this is so good.

I just want to know why my code doesn't work when I add an image to the rectangle I got blank object, you can check in my codepen here:
https://codepen.io/eacdev/pen/BaOgvYG?editors=0011

Thanks in advance guys

**Comment by ShaMan123** [comment, marked as answer]
I suggest using a [codesandbox repro](https://codesandbox.io/p/sandbox/fabric-vanillajs-sandbox-forked-v92h26?file=%2FREADME.md) instead!

**Reply by ShaMan123** [reply]
https://codesandbox.io/p/sandbox/fabric-vanillajs-sandbox-forked-jed1ru

seems that you are passing a fabric Image to pattern but it accepts a HTMLImage
I think it is good it accepts both.
